src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newElement = new Node<>(last, value, null);
+        if (size != 0) {
+            last.next = newElement;
+        } else {
+            first = newElement;
+        }
+        last = newElement;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        indexCheck(index);
+        if (index != 0) {
+            Node<T> currentNode = getNodeByIndex(index).prev;
+            Node<T> newElement = new Node<>(currentNode, value, currentNode.next);
+            currentNode.next.prev = newElement;
+            currentNode.next = newElement;
+        } else {
+            Node<T> newElement = new Node<>(null, value, first);
+            first.prev = newElement;
+            first = newElement;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T element : list) {
+            add(element);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheck(index);
+        Node<T> oldNode = getNodeByIndex(index);
+        T deletedItem = oldNode.item;
+        oldNode.item = value;
+        return deletedItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        Node<T> currentNode = getNodeByIndex(index);
+        if (currentNode.prev == null) {
+            first = currentNode.next;
+        } else {
+            currentNode.prev.next = currentNode.next;
+        }
+        if (currentNode.next == null) {
+            last = currentNode.prev;
+        } else {
+            currentNode.next.prev = currentNode.prev;
+        }
+        size--;
```

Comment:
let's move this code at separate method `unlink(Node<T> node)` and call it here and in method remove(T t)

Reply:
done